### ISSUE TITLE
Improve tutorial

### ISSUE THREAD

**Issue by GTcreyon** [GTcreyon/SM63Redux#80]
**Describe the feature you'd like**
Currently, the in-game tutorial is rather disjointed. It struggles to communicate basic mechanics such as backflips, and it introduces double-jumping *after* the first time it becomes relevant.

It might be beneficial to have short visual animations to demonstrate moves such as backflips, similar to those seen in Celeste's later chapters.

**Do you intend to contribute this yourself? If so, how do you plan on doing it?**
Not imminently. This needs more discussion first.

**Comment by RandomCatDude** [comment]
I liked the OG game's signs that have images and animations to illustrate things, defs something worth expanding upon. Maybe non-disruptive (not pausing the game) popup overlays, showing the relevant tutorial animation/text?

Could be neat to also show the necessary keyboard inputs, but that's severely complicated by the rebindable controls...

**Comment by jaschutte** [comment]
I've also seen many people struggle with the complete basics, like jumping or reading signs. New players have no way to know how to read signs without either someone telling them or reading controls. Which is hidden behind the pause menu and you need to press another unkown key to open that.

**Comment by Koopa1018** [comment]
I have some scattered thoughts:
- I don't think we need to teach backflips in t1. The player has the entire game to master the moveset--shoving everything down their gullet first thing just leads to information overload.
- Difficult to make or not, the OG's `Z` popup over signs is a good design. Immediate feedback, unobtrusive yet clear.
- Re: gradual moveset mastery, should we even require double and triple jumps in t1? Probably yes, but it's worth considering.
- I agree about visual signs. They say a picture's worth a thousand words for a reason :P

**Comment by Koopa1018** [comment]
Well, yes, but. I meant, does the _level layout_ need to require them? We could theoretically remove all the double jump cliffs and let the player get by with FLUDD and single jumps.

Not to say we _should_ do that (where else would we teach it?), but.

**Comment by GTcreyon** [comment]
> Well, yes, but. I meant, does the _level layout_ need to require them? We could theoretically remove all the double jump cliffs and let the player get by with FLUDD and single jumps.

I don't think we should be altering the basic layout of the level from OG 63.

**Comment by Koopa1018** [comment]
That makes sense.

**Comment by jaschutte** [comment]
> > Re: gradual moveset mastery, should we even require double and triple jumps in t1? Probably yes, but it's worth considering.
> 
> I think we probably should, because the level layout specifically requires it. You can't get through the first room without either a double jump or a backflip, actually.

Then we should teach the double jump, a technique far easier to pull off and a lot more intuitive.

**Comment by jaschutte** [comment]
I also like what the original game did, automatically reading the sign at the start. This could be a toggle in settings so speedrunners could disable it, and the likeliness that a new player opens the settings and disables it is low.

**Comment by Koopa1018** [comment]
The original game let you clear that cliff with only a single jump, though???
